garage/misc/git_handler.py

```diff
@@ -0,0 +1,336 @@
+"""Perform different git tasks while running experiments."""
+import importlib
+import signal
+import time
+
+from git.exc import BadName, GitCommandError
+from termcolor import colored
+
+from garage.config import GIT_REPO_URL
+from garage.config import PROJECT_PATH
+from git import Repo
+
+
+class GitHandler:
+    """Execute diverse git commands.
+
+    This class makes use of the sh package to execute git commands just like
+    in the terminal shell, with the convenience of grouping commands to perform
+    more complex tasks.
+
+    Preconditions of this class:
+        - The working directory is not in the middle of a conflict
```

Comment:
You should detect this and exit if you can

Reply:
Ok, but I haven't not found a nice way to do this like a return value from a git command or an API from GitPython, so it may take some time.